lib/credly/error.rb

```diff
@@ -0,0 +1,48 @@
+module Credly
+  class Error < StandardError
+    attr_reader :response
+
+    def self.from_response(response)
+      klass =
+        case response.status
+        when 400 then BadRequest
+        when 401 then Unauthorized
+        when 403 then Forbidden
+        when 404 then NotFound
+        when 410 then Gone
```

Comment:
suggestion: delete line 12" when 410 then Gone", I think 410 status codes are not typically used for badge template